rlberry/utils/check_gym_env.py

```diff
@@ -0,0 +1,340 @@
+    # We also allow int because the reward will be cast to float
+    assert isinstance(
+        reward, (float, int, np.float32)
+    ), "The reward returned by `step()` must be a float"
+    assert isinstance(done, bool), "The `done` signal must be a boolean"
+    assert isinstance(
+        info, dict
+    ), "The `info` returned by `step()` must be a python dictionary"
+
+
+def _check_spaces(env: gym.Env) -> None:
+    """
+    Check that the observation and action spaces are defined
+    and inherit from gym.spaces.Space.
+    """
+    # Helper to link to the code, because gym has no proper documentation
+    gym_spaces = " cf https://github.com/openai/gym/blob/master/gym/spaces/"
+
+    assert hasattr(env, "observation_space"), (
+        "You must specify an observation space (cf gym.spaces)" + gym_spaces
+    )
+    assert hasattr(env, "action_space"), (
+        "You must specify an action space (cf gym.spaces)" + gym_spaces
+    )
+
+    assert isinstance(env.observation_space, spaces.Space), (
+        "The observation space must inherit from gym.spaces" + gym_spaces
+    )
+    assert isinstance(env.action_space, spaces.Space), (
+        "The action space must inherit from gym.spaces" + gym_spaces
+    )
+
+
+# Check render cannot be covered by CI
+def _check_render(
+    env: gym.Env, warn: bool = True, headless: bool = False
+) -> None:  # pragma: no cover
+    """
+    Check the declared render modes and the `render()`/`close()`
+    method of the environment.
+    :param env: The environment to check
+    :param warn: Whether to output additional warnings
+    :param headless: Whether to disable render modes
+        that require a graphical interface. False by default.
+    """
+    render_modes = env.metadata.get("render.modes")
+    if render_modes is None:
+        if warn:
+            logger.warn(
+                "No render modes was declared in the environment "
+                " (env.metadata['render.modes'] is None or not defined), "
+                "you may have trouble when calling `.render()`"
+            )
+
+    else:
+        # Don't check render mode that require a
+        # graphical interface (useful for CI)
+        if headless and "human" in render_modes:
+            render_modes.remove("human")
+        # Check all declared render modes
+        for render_mode in render_modes:
+            env.render(mode=render_mode)
+        env.close()
+
+
+def _check_reset_seed(env: gym.Env, seed: Optional[int] = None) -> None:
+    """
+    Check that the environment can be reset with a random seed.
+    """
+    signature = inspect.signature(env.reset)
+
+    seed_param = signature.parameters.get("seed")
+    # Check the default value is None
+    if seed_param is not None and seed_param.default is not None:
+        logger.warn(
+            "The default seed argument in reset should be `None`, "
+            "otherwise the environment will by default always be deterministic"
+        )
+
+
+def check_gym_env(
```

Comment:
What prevents us from just importing this method from gym instead of copying the code for the tests? (also, we are not using the same docstring style as gym)

Reply:
I removed some of the tests from gym that were failing on our environments. I don't remember which ones but it was mostly due to some conventions and I beleived it is not necessary to change all our envs to agree with their convention. Do you agree or should we change our environments ?

Reply:
Mainly if I remember right this was some typing issues. Like an state returned by env.step() must be int, and things like that. Some of ours are float. And also some problem because of the differences between gym.spaces and rlberry.spaces.

Reply:
Ok, no problem, I agree to keep this adaptation you made!